### DISCUSSION TITLE
Possibility to register custom hooks

### DISCUSSION THREAD

**Discussion by misaon** [nuxt/nuxt#16873] [Ideas]
Currently all runtime hooks are defined statically in `nuxt.ts`. If I create a custom call to callHook, my hook name is not Typescript typed.

```ts
// @ts-expect-error todo waiting to Nuxt integrate own hook register method
nuxtApp.hooks.callHook('oxy:product:addToCart', data).catch(() => {})
```

![image](https://user-images.githubusercontent.com/15806356/190075658-a6211562-544b-4896-a270-40ae1d458bd1.png)

**Comment by danielroe** [comment]
This is actually possible now. You need to augment the RuntimeNuxtHooks interface from `#app`.

**Comment by Tarabass** [comment]
@danielroe @misaon 

I don't get this to work. I added this to my `vue-shim.d.ts` but still got the same error:

```
import { HookResult } from "@nuxt/schema";

declare module '#app' {
  interface RuntimeNuxtHooks {
    'form:submit': () => HookResult
  }
}
```

![image](https://github.com/nuxt/nuxt/assets/462513/c52d884a-3347-4ac6-8354-d844d72acef8)

**Reply by danielroe** [reply]
Try opening an issue with a reproduction

**Reply by Tarabass** [reply]
Done:

https://github.com/nuxt/nuxt/issues/26510
